merkle/hstar2.go

```diff
@@ -59,13 +59,16 @@ func NewHStar2(treeID int64, hasher hashers.MapHasher) HStar2 {
 // which contains the given set of non-null leaves.
 func (s *HStar2) HStar2Root(depth int, values []*HStar2LeafHash) ([]byte, error) {
 	sort.Sort(ByIndex{values})
-	combine := func(depth int, offset *big.Int, lhs, rhs []byte) ([]byte, error) {
+	combine := func(depth int, index *big.Int, lhs, rhs []byte) ([]byte, error) {
 		h := s.hasher.HashChildren(lhs, rhs)
 		return h, nil
 	}
 	return s.hStar2b(0, depth, values, smtZero, nil, combine)
 }
 
+// PrefetchNodeFunc reports coordinates of a Merkle tree node to prefetch.
+type PrefetchNodeFunc func(depth int, index *big.Int)
```

Comment:
Optional: You could call this PrefetchNodeVisitor if you want to give a hat tip to the design pattern it mirrors:
https://en.wikipedia.org/wiki/Visitor_pattern

Reply:
Done.